R/read_vpts.R

```diff
@@ -1,241 +1,144 @@
+
+  # Get file extension
+  extension <- unique(tools::file_ext(files))
+
+  assertthat::assert_that(
+    length(extension) == 1,
+    msg = "`files` must all have the same extension."
+  )
+
+  # If the file has an extension, check if it is valid
+  if (extension != "") {
+      assertthat::assert_that(
+      extension %in% valid_extensions,
+      msg = glue::glue(
+        "`files` must have one of the following extensions: {valid_extensions_collapse}",
+        valid_extensions_collapse = glue::glue_collapse(valid_extensions, sep = ", ")
       )
-    if (lat < -90 || lat > 90) {
-      stop("'lat' should be numeric between -90 and 90 degrees")
-    }
-  }
-  if (!missing(lon)) {
-    assertthat::assert_that(
-      assertthat::is.number(lon),
-      msg = "'lon' should be a single numeric numeric between -360 and 360 degrees"
     )
-    if (lon < -360 || lon > 360) {
-      stop("'lon' should be numeric between -360 and 360 degrees")
-    }
-  }
-  if (!missing(height)) {
-    assertthat::assert_that(assertthat::is.number(height))
-    if (!is.numeric(height) || height < 0) {
-      stop("'height' should be a positive number of meters above sea level")
-    }
-  }
-  if (missing(radar) && sep =="") {
-    stop("'radar' argument missing. Required to specify a radar identifier.")
-  }
-  if (missing(wavelength)) {
-    warning(paste("No 'wavelength' argument provided, assuming radar operates",
-      " at ", wavelength, "-band",
-      sep = ""
-    ))
+  } else {
+    # If the file does not have an extension, infer the file type
+    extension <- guess_file_type(files)
   }
-  wavelength_msg <-
-    glue::glue(
-      "'wavelength' should be a single positive number",
-      ", or one of 'C' or 'S' for C-band and S-band radar, respectively."
-    )
-  assertthat::assert_that(
-    (assertthat::is.number(wavelength) && wavelength > 0) ||
-      (assertthat::is.scalar(wavelength) && wavelength %in% c("C","S")),
-    msg = wavelength_msg)
 
-  if (wavelength == "C") {
-    wavelength <- 5.3
-  }
-  if (wavelength == "S") {
-    wavelength <- 10.6
+  # Check if the input file has a .txt extension and if so reroute to read_stdout
+  if (extension == "txt") {
+    warning(".txt extenstion detected - falling back to read_stdout(). The use of read_stdout() will be deprecated soon.
+    Please consider updating your code to use csv or h5 input files")
+    return(do.call(read_stdout, c(list(file = files, wavelength = "C", sep = ""), list(...))))
   }
 
-  # header of the data file
-  header.names.short <- c(
-    "Date", "Time", "height", "u", "v", "w", "ff", "dd",
-    "sd_vvp", "gap", "dbz", "eta", "dens", "DBZH", "n",
-    "n_dbz", "n_all", "n_dbz_all"
-  )
-  header.names.long <- c(
-    "Date", "Time", "height", "u", "v", "w", "ff", "dd",
-    "sd_vvp", "head_bl", "head_ff", "head_dd", "head_sd",
-    "gap", "dbz", "eta", "dens", "DBZH", "n", "n_dbz",
-    "n_all", "n_dbz_all"
+  # Read files
+  data <- switch(extension,
```

Comment:
suggest renaming `extension` to `file_type` since we will allow text files without extension